src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -71,7 +71,12 @@ public function getProxyFactoryCode(Definition $definition, $id)
             $instantiation .= " \$this->services['$id'] =";
         }
 
-        $methodName = 'get'.Container::camelize($id).'Service';
+        if (func_num_args() >= 3) {
+            $methodName = func_get_arg(2);
```

Comment:
hmm, actually no. It would break BC for people extending the class. so forget it.